visualization/app/codeCharta/codeCharta.service.spec.ts

```diff
@@ -183,5 +187,28 @@ describe("codeChartaService", () => {
 					done()
 				})
 		})
+
+		it("should migrate old attribute types", done => {
+			validFileContent.attributeTypes = {
+				nodes: [{ rloc: AttributeTypeValue.absolute }],
+				edges: [{ pairing: AttributeTypeValue.relative }]
+			}
+
+			codeChartaService
+				.loadFiles([
+					{
+						fileName,
+						content: validFileContent
+					}
+				])
+				.then(() => {
+					const attributeTypes: AttributeTypes = {
+						nodes: { rloc: AttributeTypeValue.absolute },
+						edges: { pairing: AttributeTypeValue.relative }
+					}
+					expect(getCCFiles(storeService.getState().files)[0].settings.fileSettings.attributeTypes).toEqual(attributeTypes)
+					done()
```

Comment:
Instead of using `done`, it should be possible to return a promise. Using an async function would be ideal.